File: MedContactApp/Models/DoctDataIndexViewModel.cs
using MedContactApp.FilterSortHelpers;
using MedContactApp.FilterSortPageHelpers;
using MedContactCore.DataTransferObjects;
using MedContactDb.Entities;

namespace MedContactApp.Models
{
    public class DoctDataIndexViewModel
    {
        public IEnumerable<DoctorFullDataDto> Users { get; }
        public PageViewModel PageViewModel { get; }
        public FilterSpecViewModel FilterViewModel { get; }
        public SortViewModel SortViewModel { get; }
        public string ProcessOptions { get; } = string.Empty;
        public DoctDataIndexViewModel (IEnumerable<DoctorFullDataDto> users,
            string processOptions, PageViewModel pageViewModel,
            FilterSpecViewModel filterViewModel, SortViewModel sortViewModel)
        {
            Users = users;
            ProcessOptions = processOptions;
            PageViewModel = pageViewModel;
            FilterViewModel = filterViewModel;
            SortViewModel = sortViewModel;
        }
    }
}

File: medcontact-angular/medcontact/src/app/models/doctdataindexviewmodel.ts
import { Doctorfulldata } from './doctorfulldata';
import { Filterspecviewmodel } from './filterspecviewmodel';
import { Pageviewmodel } from './pageviewmodel';
import { Sortviewmodel } from './sortviewmodel';

export class Doctdataindexviewmodel {
 users?: Doctorfulldata[];
 pageViewModel?: Pageviewmodel;
 filterViewModel?: Filterspecviewmodel;
 sortViewModel?: Sortviewmodel;
 processOptions?: string;
 reflink?: string;
}
